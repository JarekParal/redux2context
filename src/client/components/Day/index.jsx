/* @flow */
import withIntl from "client/services/intl/withIntl";
import { DEFAULT_DATE } from "client/services/intl/consts";

type Props = {
  date: Date,
  locale: string,
  format?: { [key: string]: number | string },
};

const Day = (props: Props) =>
  new Intl.DateTimeFormat(props.locale, props.format).format(props.date);

Day.defaultProps = {
  format: DEFAULT_DATE,
};

export default withIntl(Day);
