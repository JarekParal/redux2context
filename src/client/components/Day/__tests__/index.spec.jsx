/* eslint-disable flowtype/no-types-missing-file-annotation */
import * as React from "react";
import { shallow } from "enzyme";

import Day from "../index";

describe("#Day", () => {
  it("should render correctly", () => {
    const date = new Date(2017, 9, 28);
    const wrapper = shallow(<Day date={date} />);

    expect(wrapper).toMatchSnapshot();
  });
});
