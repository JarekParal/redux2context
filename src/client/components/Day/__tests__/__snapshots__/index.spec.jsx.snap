// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Day should render correctly 1`] = `
<Day
  date={2017-10-27T22:00:00.000Z}
  format={
    Object {
      "day": "numeric",
      "month": "numeric",
      "year": "numeric",
    }
  }
  locale="en"
  translate={[Function]}
/>
`;
